src/caffe/solver.cpp

```diff
@@ -211,7 +228,9 @@ void Solver<Dtype>::Step(int iters) {
       losses[idx] = loss;
     }
     if (display) {
-      LOG(INFO) << "Iteration " << iter_ << ", loss = " << smoothed_loss;
+      if (Caffe::root_solver()) {
```

Comment:
probably a bit late to comment on this, and to me not necessary for merge, but these conditional `LOG(INFO)` calls could be made a bit more compact using `LOG_IF`s, e.g. `LOG_IF(INFO, Caffe::root_solver()) << "Iteration..."`